lib/plugins/blocks.js

```diff
@@ -454,6 +463,55 @@ function inject (bot, { version, storageBuilder }) {
     })
   }
 
+  /**
+   * Places a sign and writes to it
+   * @throws {Error} Throws error when no sign is equipped or the position to place the sign is is already occupied
+   * @param {import('vec3').Vec3} pos
+   * @param {string} text
+   * @param { { sneak?: boolean, writeDelay?: number, forceLook?: boolean } } options
+   * @returns {Promise<void>}
+   */
+  bot.placeSign = async (pos, text, options = {}) => {
+    if (!(pos instanceof Vec3)) throw new Error('Invalid argument for pos. Must be a instance of Vec3 got ' + typeof pos)
+    if (!bot.heldItem || (bot.heldItem && !bot.heldItem.name.includes('sign'))) throw new Error('Need a sign equipped to place sign')
+    const signBlock = bot.blockAt(pos)
+    if (!signBlock || !signBlock?.position || (signBlock && !signBlock.name.includes('air'))) throw new Error('Invalid sign position')
+    const forceLook = options.forceLook ?? true
+    const groundPlacable = bot.blockAt(signBlock.position.offset(0, -1, 0))?.boundingBox === 'block'
+    let wallPlacement = null
+    const directions = [new Vec3(1, 0, 0), new Vec3(0, 0, 1), new Vec3(-1, 0, 0), new Vec3(0, 0, -1)]
+    for (const dir of directions) {
+      const tmp = bot.blockAt(pos.minus(dir))
+      if (tmp?.boundingBox === 'block') {
+        wallPlacement = dir
+        break
+      }
+    }
+
+    if (!groundPlacable && wallPlacement === null) throw new Error('Cannot place block in position ' + pos.toString())
```

Comment:
This is suppose the check for potential wall placement spots. If it dose not find a ground placement block or any off the wall placement blocks it will throw an error. There is not way to specify where you want to place the sign. The function looks for a spot for you. I can maybe add a allowWallPlacement and allowFloorPlacement option.

Reply:
I think it should be explicit and well defined here. Maybe seperate functions to place a wall or standing sign would make things clear, which can prefill options to placeSign. A bit like currently placing blocks